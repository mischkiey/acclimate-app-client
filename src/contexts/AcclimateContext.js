import React from 'react';

const AcclimateContext = React.createContext({});

export default AcclimateContext;