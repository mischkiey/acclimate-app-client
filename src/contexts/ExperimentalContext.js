import React, { Component } from 'react';
import TokenService from '../services/token-services';

export const ExperimentalContext = React.createContext({});

export class ExperimentalProvider extends Component {
    state = {
        log: false,
        tasks: [],
        shoppingItems: [],
    };

    handleUserLog = () => {
        const log = TokenService.hasAuthToken();
        this.setState({log});
    };

    handleUserLogOut = () => {
        TokenService.clearAuthToken();
        this.handleUserLog();
    };

    setTasks = (tasks) => {
        this.setState({tasks})
    };

    setShoppingItems = (shoppingItems) => {
        this.setState({shoppingItems})
    };

    render() {
        const value = {
            log: this.state.log,
            tasks: this.state.tasks,
            shoppingItems: this.state.shoppingItems,

            handleUserLog: this.handleUserLog,
            handleUserLogOut: this.handleUserLogOut,
            setTasks: this.setTasks,
            setShoppingItems: this.setShoppingItems,
        };

        return (
            <ExperimentalContext.Provider value={value}>
                {this.props.children}
            </ExperimentalContext.Provider>
        );
    };
};

