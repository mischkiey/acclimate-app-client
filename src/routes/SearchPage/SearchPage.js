import React, { Component } from 'react';
import SearchForm from '../../components/SearchForm/SearchForm';

class SearchPage extends Component {
    render() {
        return (
            <SearchForm />
        );
    };
};

export default SearchPage;