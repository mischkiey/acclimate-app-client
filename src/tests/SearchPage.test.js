import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { AcclimateProvider } from '../contexts/AcclimateContext';
import SearchPage from '../routes/SearchPage/SearchPage';

describe(`SearchPage Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <AcclimateProvider>
                        <SearchPage />
                    </AcclimateProvider>
                </BrowserRouter>,
                div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <AcclimateProvider>
                            <SearchPage />
                        </AcclimateProvider>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})