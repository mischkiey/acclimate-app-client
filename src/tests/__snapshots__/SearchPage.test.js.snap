// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage Component Snapshot test Renders the UI as expected 1`] = `
Array [
  <h2
    className="center"
  >
    Select a Disaster Program
  </h2>,
  <form
    className="search-form-group"
  >
    
    <div
      className="search-form-inputs-group"
    />
  </form>,
]
`;
